### PR TITLE
Adds implementation for railcom messages related to logon

### PR DESCRIPTION
**Pull request by balazsracz**
- Adds constants for the automatic logon railcom messages
- Adds helper functions in the RailcomDefs to generate the S-9.2.1.1 / RCN-218 feedback messages.
- Adds a processor for a railcom hub that recognizes these feedback messages and checks for correctness.

### CODE REVIEW

File: src/dcc/LogonFeedback.hxx [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+                break;
+            case PacketType::LOGON_ENABLE:
+                any_error |=
+                    (((data >> 44) & 0xf) != RMOB_LOGON_ENABLE_FEEDBACK);
+                cb_->process_decoder_id(key, any_error, data & PAYLOAD_MASK);
+                break;
+            case PacketType::GET_DATA_START:
+            case PacketType::GET_DATA_CONT:
+            case PacketType::MISC_254:
+            case PacketType::UNKNOWN:
```
Comment (bakerstu) [inline src/dcc/LogonFeedback.hxx:149]: Do you want to add a default case here?

File: src/dcc/LogonFeedback.hxx [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+        else
+        {
+            data |= ERROR_UNKNOWN;
+        }
+        shift -= 6;
+    }
+
+    /// Parses a concatenated ch1+ch2 response into a single uint64_t. It
+    /// contains error flags in the MSB, the number of valid 6-bit codepoints
+    /// in the second MSB , and 6 bytes data in the LSB, entered MSB-first.
```
Comment (bakerstu) [inline src/dcc/LogonFeedback.hxx:218]: MSB first means wire order?

File: src/utils/Crc.cxxtest [hunk 3]
```diff
@@ -175,3 +175,25 @@ TEST(Crc8Test, Collision)
     LOG(INFO, "s1=%s s2=%s", string_to_hex(s1).c_str(),
         string_to_hex(s2).c_str());
 }
+
+TEST(Crc8Test, LogonExample)
```
Comment (bakerstu) [inline src/utils/Crc.cxxtest:179]: I'm a bit confused. These tests, as well as some above, don't have any "EXPECTs". What is the acceptance criteria? Doesn't crash?

**Review by bakerstu — APPROVED** [review]

File: src/dcc/LogonFeedback.hxx (hunk 1 above)
Comment (balazsracz) [reply src/dcc/LogonFeedback.hxx:149]: done

File: src/dcc/LogonFeedback.hxx (hunk 2 above)
Comment (balazsracz) [reply src/dcc/LogonFeedback.hxx:218]: clarified:
```
    /// Parses a concatenated ch1+ch2 response into a single uint64_t. It
    /// contains error flags in the MSB, the number of valid 6-bit codepoints
    /// in the second MSB , and 6 bytes data in the LSB, entered with the first
    /// wire byte in the third MSB and last wire byte in the LSB.
    ///
    /// Any input byte that was missing or contained any other codepoint than a
    /// valid 6-bit data codepoint (including garbage or ACK) is translated
    /// into zero bits in the parse result.
    /// @param fb feedback to parse
    /// @return parse result.
```

File: src/utils/Crc.cxxtest (hunk 3 above)
Comment (balazsracz) [reply src/utils/Crc.cxxtest:179]: added comment:
```
// These examples contain specific messages from other unittests and print
// their CRC. This pattern can be used for developing other tests. There are no
// expectations in these tests.
```

**Pull request merged**